presto-ranger-plugin/src/main/java/io/prestosql/ranger/RangerSystemAccessControlFactory.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.ranger;
+
+import com.google.common.base.Strings;
+import io.airlift.log.Logger;
+import io.prestosql.ranger.groups.EmptyUserGroups;
+import io.prestosql.ranger.groups.UserGroups;
+import io.prestosql.ranger.groups.UserGroupsRangerClient;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.StandardErrorCode;
+import io.prestosql.spi.security.SystemAccessControl;
+import io.prestosql.spi.security.SystemAccessControlFactory;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.ranger.audit.provider.MiscUtil;
+import org.apache.ranger.authorization.hadoop.config.RangerConfiguration;
+import org.apache.ranger.plugin.audit.RangerDefaultAuditHandler;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+public class RangerSystemAccessControlFactory
+        implements SystemAccessControlFactory
+{
+    protected static final String RANGER_ACCESS_CONTROL = "ranger-access-control";
+    protected static final String RANGER_GROUP_FETCH = "ranger-groups-fetch";
+    private static final Logger log = Logger.get(RangerSystemAccessControlFactory.class);
+
+    @Override
+    public String getName()
+    {
+        return RANGER_ACCESS_CONTROL;
+    }
+
+    @Override
+    public SystemAccessControl create(Map<String, String> config)
+    {
+        RangerConfiguration rangerConfig = RangerConfiguration.getInstance();
```

Comment:
You should store away the pluginClassLoader in constructor and use it in create() method.

We can follow the example of ClassLoaderSafeConnectorMetadata to create a wrapper on top of RangerSystemAccessControl to use pluginClassloader for all SystemAccessControl functions.

Reply:
Acked